src/utils/useGeolocation.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+
+interface locationType {
+  loaded: boolean;
+  coordinates?: { lat: number; lng: number };
+  error?: { code: number; message: string };
+}
+
+const useGeolocation = () => {
```

Comment:
그리고 화살표 함수말고, funtion을 써주세요
```suggestion
funtion useGeolocation(){
```